src/uwtools/drivers/forecast.py

```diff
@@ -126,10 +131,64 @@ def resources(self) -> None:
         ???
         """
 
-    def run(self):
+    def run(
+        self,
+        config_file: str,
+        forecast_app: str,
+        forecast_model: str,
+        dry_run: bool = False,
+        log_file: Optional[str] = None,
+        machine: Optional[str] = None,
+        quiet: bool = False,
+        verbose: bool = False,
+    ) -> None:  # pragma: no cover
         """
         Runs FV3.
         """
+        # set up logging
+        name = f"{inspect.stack()[0][3]}"
+        log = cli_helpers.setup_logging(
+            log_file=log_file or "/dev/stdout",
+            log_name=name,
+            quiet=quiet,
+            verbose=verbose,
+        )
+        # read in the config file
+        # self.create_model_config(config_file=config_file)
+        experiment_config = config.YAMLConfig(config_file)
+        # prepare directories
+        run_directory = experiment_config["parameters"]["run_directory"]
+        self.create_directory_structure(run_directory)
+
+        static_files = experiment_config["locations"]["static"]
```

Comment:
Again, perhaps the configuration section under `FV3Forecast` (name pending) looks something like:

```
FV3Forecast:
  run_directory:
  static:
  cycledep:
  exec_name:
platform:
  mpi_cmd:
  
```

Please also disregard my lack of ordering there.

Reply:
Maybe
```
forecast:
    model: FV3
    <etc>
```
? This avoids leaking Python-code names into the DSL. We can map to the correct class with e.g.
```
forecast_class = {"FV3": FV3Forecast, "MPAS": MPASForecast}[config["forecast"]["model"]
```
(I think we're doing that already, just constructing the class name with an f-string or similar.)

It terms of schema, I think it would be nice to say "There's an optional top-level block called `forecast` and, if it exists, it may have a `model` value in the enum `[FV3, MPAS]`, and also ..." vs having to say "There's an optional top-level block called `FV3Forecast` and another optional top-level block called `MPASForecast` and they are mutually exclusive, and their subtrees look like ..." -- with the names of Python classes also appearing in YAML.

Unless we can describe all the pertinent forecast parameters for both (or more) models with a consistent set of key-value pairs, we might have to overs-specify the parameters under `forecast` and check them in code post schema validation, unless JSON Schema permits saying things like "parameter `x` only goes with `model: FV3`, and parameter `y` only goes with `model: MPAS`. I don't know yet. I think it'd be worth trying to distill the common parameters and then complicating however we must from there.

Failing any/all of that, I'd suggest that we at least choose and stick to a convention for key strings, e.g. "lower-case with hyphens". (Underscores are ok, too, but I've found value in using hyphens to remind everyone that these things are _not anything except YAML config keys_ Specifically, they're _not_ Python variables! For the same reason, I suggest using e.g. `true` instead of `True` in YAML: As a reminder that these are YAML values, not Python values -- as otherwise people supply `None` instead of `null` as a YAML value and get `"None": str` in Python instead of `None: NoneType`. Schema checking will help, but conventions help, too. And we can map from our DSL keys to whatever we need for various uses.)

I'm not suggesting any restriction on _value_ case, punctuation, whitespace, etc. -- whatever's needed is fine. Just on keys.

Reply:
I've gone through some of the SRW YAML files and here's my proposed edit, with some similar naming and structure to the others:
```
 platform:
    RUN_ENVIR: community
    MACHINE: hera
    MPI_CMD: srun
    ACCOUNT: any_acct
forecast:
    APP: SRW
    EXTRN_MDL_NAME: FV3GFS
    STATIC:
    CYCLEDEP:
    VERBOSE: false
```
This should formally pull the args out to the config file as well